src/minefield.rs

```diff
@@ -163,9 +206,39 @@ impl Board {
     /// starting from a player-uncovered tile
     /// and stopping on already uncovered tiles and hint tiles.
     fn flood_uncover(&mut self, x: usize, y: usize) {
-        todo!()
+        let mut tile_pos = VecDeque::new();
+        tile_pos.push_back((x, y));
+
+        while let Some(tile) = tile_pos.pop_front() {
+            let t_idx = self.coords_to_index(tile.0, tile.1);
+            if self.tiles[t_idx].is_uncoverable() && !self.tiles[t_idx].is_mine() {
+                self.tiles[t_idx].cover = Cover::Down;
+                if self.covered > self.params.mines {
+                    self.covered -= 1;
+                }
+
+                if !self.tiles[t_idx].is_hint() {
+                    let min_x = if tile.0 > 0 { tile.0 - 1 } else { 0 };
+                    let max_x = if tile.0 < self.params.width { tile.0 + 1 } else { tile.0 };
+                    let min_y = if tile.1 > 0 { tile.1 - 1 } else { 0 };
+                    let max_y = if tile.1 < self.params.height { tile.1 + 1 } else { tile.1 };
+
+                    for i in min_x..=max_x {
+                        for j in min_y..=max_y {
+                            let idx = self.coords_to_index(i, j);
+                            if self.tiles[idx].is_uncoverable() {
+                                tile_pos.push_back((i, j));
+                            }
+                        }
+                    }
+                }
+            }
+        }
     }
 
+    /// Get single dimension index of 2D tile
+    ///
+    /// This method exists purely because borrow checker took me hostage :)
```

Comment:
This comment doesn't seem necessary anymore :P

Reply:
:) Done